### ISSUE TITLE
Is there a reason why database table ids are of type CHAR(36)?

### ISSUE THREAD

**Issue by wayoda** [codeschluss/wupportal#1]
Hi, 
I'm just browsing through the backend code. 
Is there a reason why all table-ids are define as `CHAR(36)`. Since the value of the `id` itself doesn't seem to have specific meaning for the application, wouldn't  it be easier to use the `MYSQL-Autoincrement` feature:

https://dev.mysql.com/doc/refman/5.7/en/example-auto-increment.html

In the current code the application has to provide a unique string 
```sql
CREATE TABLE `configurations` (
    `id` CHAR(36) NOT NULL PRIMARY KEY,

INSERT INTO `configurations` (`id`, `item`, `value`) VALUES
    ('config1', 'mapcenterLongitude', '7.1756'),
```
With a simple integer key the database system will provide a unique key for each row int the tables
```sql
CREATE TABLE `configurations` (
     id INT NOT NULL AUTO_INCREMENT,
     PRIMARY KEY (id)

INSERT INTO `configurations` (`item`, `value`) VALUES
    ('mapcenterLongitude', '7.1756'),
```
and it takes care to create that unique id for you on each INSERT statement (value for `id`can be omitted from the INSERT).

cheers
Eberhard

**Comment by schlusslicht** [comment]
Hello Eberhard,
first, let me thank you for your interest and response.

We decided to rely on CakePHPs feature to generate UUIDs when inserting records, therefore the table definitions feature `CHAR(36)` instead of an `AUTO_INCREMENT`ed `INT`. From the [CakePHP 3.x documentation](https://book.cakephp.org/3.0/en/intro/conventions.html#model-and-database-conventions):

> In addition to use an auto-increment key as the primary key, you may also use UUID columns. CakePHP will create a unique 36 character UUID (`Cake\Utility\Text::uuid()`) whenever you save a new record using the `Table::save()` method.

For the sake of some internal mappings we chose to use UUIDs, as they are collision-save, instead of integer keys.

Best regards
Phil
